### PR TITLE
Add ApiScan to nightly build

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/25351/APIScan-step-by-step-guide-to-setting-up-a-Pipeline

The ApiScan task has been added to the nightly build and test run.  This
task should help us identify related issues earlier, rather than having
to wait for a full scan of VS.  The task can take a long time to execute
so it has been added to the nightly job rather than the PR or CI jobs.

### CODE REVIEW

File: build-tools/automation/azure-pipelines-nightly.yaml [hunk 1]
```diff
@@ -280,3 +280,55 @@ stages:
...
+
+    ### Copy .dll and .pdb files for APIScan
+    - task: CopyFiles@2
+      displayName: Collect Files for APIScan
+      inputs:
+        Contents: $(System.DefaultWorkingDirectory)\bin\$(XA.Build.Configuration)\dotnet\packs\Microsoft.Android*\**\?(*.dll|*.pdb)
+        TargetFolder: $(Build.StagingDirectory)\apiscan
+        OverWrite: true
+        flattenFolders: true
+      condition: and(succeeded(), eq(variables['ApiScan.Enabled'], 'true'), eq(variables['Build.SourceBranch'], 'refs/heads/dev/pjc/nightly-apiscan'))
```
Comment (jpobst) [inline build-tools/automation/azure-pipelines-nightly.yaml:316]: I suspect this is leftover from testing: `refs/heads/dev/pjc/nightly-apiscan`.

**Review by jpobst — COMMENTED** [review]
Do you have a link to a test build that shows successful scan output?

File: build-tools/automation/azure-pipelines-nightly.yaml (hunk 1 above)
Comment (pjcollins) [reply build-tools/automation/azure-pipelines-nightly.yaml:316]: Thanks, forgot to update these test conditions

**Comment by pjcollins** [comment]
Set up a new nightly test run here: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8864982&view=results

**Comment by pjcollins** [comment]
> Where do the results get published, and should we fail the build somehow?

Looks like the logging and build break step additions worked as expected, I think this should be good to merge now and we can iterate on fixing issues / adding exceptions.

**Review by jpobst — APPROVED** [review]

**Pull request merged**